### PR TITLE
Improve correctness of 1.8 item varieties data

### PR DESCRIPTION
**Pull request by kaduvert**

### CODE REVIEW

**Comment by kaduvert** [comment]
i abandoned the idea of doing this for all versions, it's just too complicated and i've got no clue about them.
if anyone needs exactly what i need for 1.8, i leave some scripts here which i made for data manipulation / extraction:
```js
// add block variations to item variations if they aren't present
// put in /data/pc
const fs = require('fs')

const versions = [
    '1.7',
    '1.8',
    '1.9',
    '1.10',
    '1.11',
    '1.12'
]

const itemFiles = versions.map((v) => require(`./${v}/items.json`))
const blockFiles = versions.map((v) => require(`./${v}/blocks.json`))

itemFiles.forEach((itemFile, i) => {
    itemFile.forEach(item => {
        const block = blockFiles[i].find(block => item.id === block.id)
        if (!item.variations && block && block.variations) {
            item.variations = block.variations
        }
    })
})

itemFiles.forEach((itemFile, i) => {
    fs.writeFileSync(`./${versions[i]}/items.json`, JSON.stringify(itemFile))
})
```
```js
// spawn egg name extraction
// put in /data/pc/${version}
const entityFile = require(`./entities.json`)

const metadatas = []

entityFile.sort((a, b) => a.internalId - b.internalId).forEach((entity) => {
    metadatas.push({
        metadata: entity.internalId,
        displayName: 'Spawn ' + entity.displayName
    })
})

console.log(metadatas)
```

**Comment by kaduvert** [comment]
for anyone wondering: i removed the flower pot item varieties because they don't exist; the data was faulty

File: data/pc/1.8/items.json [hunk 1]
```diff
@@ -2307,65 +3323,7 @@
     "id": 390,
     "displayName": "Flower Pot",
     "name": "flower_pot",
-    "stackSize": 64,
-    "variations": [
```
Comment (rom1504) [inline data/pc/1.8/items.json:2311]: why was this removed ?

**Comment by rom1504** [comment]
looks reasonable except for the flower pot

**Comment by kaduvert** [comment]
i answered that in the comment above...: https://github.com/PrismarineJS/minecraft-data/pull/772#issuecomment-1722413049

as you're probably not willing to verify yourself, look: https://www.youtube.com/watch?v=tWyohVt3zZo

**Comment by rom1504** [comment]
ok
